x-pack/plugins/integration_assistant/server/util/processors.ts

```diff
@@ -5,24 +5,44 @@
  * 2.0.
  */
 
-import type { ESProcessorItem, Pipeline } from '../../common';
+import { load } from 'js-yaml';
+import { join as joinPath } from 'path';
+import { Environment, FileSystemLoader } from 'nunjucks';
 import { deepCopy } from './util';
+import type { ESProcessorItem, Pipeline } from '../../common';
 
 export function combineProcessors(
   initialPipeline: Pipeline,
-  processors: ESProcessorItem[]
+  processors: object[],
+  graphType: string
 ): Pipeline {
   // Create a deep copy of the initialPipeline to avoid modifying the original input
   const currentPipeline = deepCopy(initialPipeline);
-
-  // Add the new processors right before the last 2 removeprocessor in the initial pipeline.
+  if (Object.keys(processors).length === 0) {
+    return currentPipeline;
+  }
+  // Add the new processors right before the last 2 remove processor in the initial pipeline.
   // This is so all the processors if conditions are not accessing possibly removed fields.
   const currentProcessors = currentPipeline.processors;
+  const appendProcessors = createAppendProcessors(processors, graphType);
   const combinedProcessors = [
     ...currentProcessors.slice(0, -2),
-    ...processors,
+    ...appendProcessors,
     ...currentProcessors.slice(-2),
   ];
   currentPipeline.processors = combinedProcessors;
   return currentPipeline;
 }
+
+// The related and categorization graphs returns a simplified array of append processors.
+// This function converts the simplified array to the full ESProcessorItem array.
+function createAppendProcessors(processors: object[], graphType: string): ESProcessorItem[] {
+  const templatesPath = joinPath(__dirname, '../templates/processors');
+  const env = new Environment(new FileSystemLoader(templatesPath), {
+    autoescape: false,
+  });
+  const template = env.getTemplate(`${graphType}.yml.njk`);
```

Comment:
Do we expect the produced processor blocks to actually be different for related vs categorization steps? 

I tried looking at the two templates, and I see the difference, but don't see why 
1) the value field for related is enclosed in brackets, but not for categorization
2) the backup if condition null check would not also be useful for related

Reply:
I agree that this specific implementation is a bit hacky, I will remove the need to pass graphType directly, and pass it in the processors object instead, and merge the two files into a single append.yml.njk instead of having one with graph names.

However I do want to keep them as two separate objects in the template file mostly for readability. 
They do indeed have a few different rules and guidelines, both from the expected LLM output and how the append processors will look like.
**Related:**
LLM should only output the field and the name of the "value_field", which is the path to the field we want to copy the value from.
To prevent as much recursion errors as possible, I have removed anything related to "if" statements from the LLM, and expect it only to return these two values.
Before, we expected the LLM to return the value encapsulated in triple brackets for the ingest pipeline templating values, and an if condition which was always the same (the null check). It could sometimes get the brackets wrong.
Since almost all our related.* append processors are identical, it was easier to auto-generate the nullcheck if statement based on value_field, and ignore any potential if statement from the LLM (though it should not return them anymore anyway).

Example output from LLM:
Example output:
```json
[{
      "field": "related.hosts",
      "value_field": "vulnerability.reference"
    }]
```

**Categorization**
Categorization unlike related can include if conditions from the LLM, but we also allow no if condition as well, which would be hard to check for without having plenty of nested if statements in the template if we tried to do it all in one object template.
The LLM can also return one or more values and returns the square brackets itself. The related should only have 1 value at any point, and does not need or should use square brackets.

Example output:
```json
[{
      "field": "event.type",
      "value": [
        "info"
      ]
    },
    {
      "field": "event.category",
      "value": [
        "network"
      ],
      "if": "ctx.vulnerability?.description?.contains('remote')"
    }]
```

Reply:
Thanks for the detailed explanation, I understand now